### DISCUSSION TITLE
Mesh Central Failure after update

### DISCUSSION THREAD

**Discussion by andyoakeley** [amidaware/tacticalrmm#1721] [General]
Hi,

I have just updated to 0.17.3 (from i think 0.17.1) and Mesh Central is now not starting.
Everything else appears to be working fine.

```
tactical@rmm:/meshcentral/meshcentral-data$ sudo systemctl status --full --no-pager meshcentral
meshcentral.service - MeshCentral Server
     Loaded: loaded (/etc/systemd/system/meshcentral.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Fri 2023-12-29 23:38:56 AWST; 3s ago
    Process: 11126 ExecStart=/usr/bin/node node_modules/meshcentral (code=exited, status=1/FAILURE)
   Main PID: 11126 (code=exited, status=1/FAILURE)
        CPU: 133ms

Dec 29 23:38:56 rmm systemd[1]: meshcentral.service: Main process exited, code=exited, status=1/FAILURE
Dec 29 23:38:56 rmm systemd[1]: meshcentral.service: Failed with result 'exit-code'.
```

Update process was as per https://docs.tacticalrmm.com/update_server/ 
- before doing the tactical update I ran sudo apt update and sudo apt -y upgrade
- and certificate was also updated (and re-ran the update with -force after doing the cert update)

I have been through the troubleshooting here: https://docs.tacticalrmm.com/troubleshooting/
- In this it states to check mongod, but mongod not installed? Suspect this is old documentation as everything else seems to be working OK and /meshcentral-data/config.json has a section for postgres not mongodb

I have run the troubleshooting script, and this looks a bit interesting, but unsure if it is related to the MeshCentral issue:
(happy to private message the full checklog.log or other log files if needed)
```
 Getting summary output of logs
           ^^^^^^^^^^^^^^^^^^^^^
  File "/rmm/api/env/lib/python3.11/site-packages/django/db/backends/postgresql/base.py", line 275, in get_new_connection
    connection = self.Database.connect(**conn_params)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/rmm/api/env/lib/python3.11/site-packages/psycopg/connection.py", line 738, in connect
    raise ex.with_traceback(None)
django.db.utils.OperationalError: connection failed: :1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?
```

Running on Debian 12
Have also tried rebooting the machine.
Can roll back to a snapshot prior to the updates if needed.

Any tips on next steps please?

Ta

Andy

**Comment by wh1te909** [comment]
my guess is related to this https://github.com/amidaware/tacticalrmm/discussions/1720#discussioncomment-7969935
what is the output of `which npm` ? I'm guessing it will return nothing

**Reply by andyoakeley** [reply]
Indeed... you are correct. `which npm` returns nothing.

**Reply by wh1te909** [reply]
thx. i found the issue, debian repo now has node 18 and overriding nodesource and removing npm. I just pushed a fix for this [here](https://github.com/amidaware/tacticalrmm/commit/df3e68fbaf29f9018d81b184134e5b7c74661ab1) so please go ahead and run the update script again with the force flag `./update.sh --force` (the update script should self update itself) and it should be fixed.

**Reply by andyoakeley** [reply]
Thanks. That's working now.
Thanks for the swift response, especially at this time of year.
